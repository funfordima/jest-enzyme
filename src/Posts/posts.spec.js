import React from "react";
import Posts from "./posts";

describe('Post component', () => {
  it('should render Posts component', () => {
    const component = shallow(<Posts />);

    expect(component).toMatchSnapshot();
  });

  it('should render Posts component with children', () => {
    const component = render(<Posts />);

    expect(component).toMatchSnapshot();
  });
});