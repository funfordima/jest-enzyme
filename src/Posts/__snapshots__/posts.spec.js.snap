// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render Posts component 1`] = `
<ul
  className="newsList"
>
  <Post
    author="Dmytro"
    created_at="2020-05-03T23:36:09.816Z"
    key="1"
    num_comments={10}
    points={100}
    title="Jest & Enzyme"
    url="//test.url"
  />
  <Post
    author="Stepan"
    created_at="2020-05-05T23:36:09.816Z"
    key="2"
    num_comments={8}
    points={10}
    title="TypeScript Basics"
    url="//test2.url"
  />
</ul>
`;

exports[`Post component should render Posts component with children 1`] = `
<ul
  class="newsList"
>
  <li
    class="post"
  >
    <div
      class="description"
    >
      <a
        class="title"
        href="//test.url"
      >
        Jest & Enzyme
      </a>
      <span
        class="text"
      >
        100 points
      </span>
      <span
        class="comments"
      >
        10 comments
      </span>
      <span
        class="date"
      >
        5/4/2020
      </span>
      <span
        class="author"
      >
        Dmytro
      </span>
    </div>
  </li>
  <li
    class="post"
  >
    <div
      class="description"
    >
      <a
        class="title"
        href="//test2.url"
      >
        TypeScript Basics
      </a>
      <span
        class="text"
      >
        10 points
      </span>
      <span
        class="comments"
      >
        8 comments
      </span>
      <span
        class="date"
      >
        5/6/2020
      </span>
      <span
        class="author"
      >
        Stepan
      </span>
    </div>
  </li>
</ul>
`;
